web-build/Dockerfile.frankenphp

```diff
@@ -0,0 +1,74 @@
+#ddev-generated
+COPY --from=frankenphp /usr /usr
+COPY --from=frankenphp /etc/frankenphp /etc/frankenphp
+RUN ldconfig
+ARG PHP_INI_DIR=/usr/local/etc/php
+ARG PHP_CFLAGS="-fstack-protector-strong -fpic -fpie -O2 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64"
+ARG PHP_CPPFLAGS="-fstack-protector-strong -fpic -fpie -O2 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64"
+
+# Don't run cleanup in install-php-extensions, because it removes build tools,
+# and running install-php-extensions second time later is going to fail because of that.
+# So we just remove the cleanup step from the script.
+RUN sed -i '/^cleanup$/d' /usr/local/bin/install-php-extensions
+
+ARG FRANKENPHP_DEFAULT_EXTENSIONS=""
+
+ARG FRANKENPHP_CUSTOM_EXTENSIONS=""
+
+# Install default extensions if not empty
+RUN if [ -n "${FRANKENPHP_DEFAULT_EXTENSIONS}" ]; then \
+        for ext in ${FRANKENPHP_DEFAULT_EXTENSIONS}; do \
+            install-php-extensions "$ext" || true; \
+        done; \
+    fi
+
+# Install user-defined extensions if any
+RUN if [ -n "${FRANKENPHP_CUSTOM_EXTENSIONS}" ]; then \
+        install-php-extensions ${FRANKENPHP_CUSTOM_EXTENSIONS}; \
+    fi
+
+RUN <<EOF
+set -eu -o pipefail
+chmod -fR ugo+w /usr/sbin /usr/bin /usr/local/etc/php
+
+rm -f /usr/local/etc/php/conf.d/docker-php-ext-xdebug.ini /usr/local/etc/php/conf.d/docker-php-ext-xhprof.ini
+
+cat <<FILE_EOF > /usr/local/bin/enable_xdebug
+#!/usr/bin/env bash
+cp /etc/php/\${DDEV_PHP_VERSION}/mods-available/xdebug.ini /usr/local/etc/php/conf.d/docker-php-ext-xdebug.ini
+supervisorctl restart webextradaemons:frankenphp >/dev/null || true
+echo "Enabled xdebug"
+FILE_EOF
```

Comment:
I think it would be clearer and easier to maintain to provide these as files and use ADD with them. But you already thought of that and decided against...